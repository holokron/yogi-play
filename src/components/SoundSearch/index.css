.input-rounded {
  border-radius: 30px 30px 30px 30px !important;
}
