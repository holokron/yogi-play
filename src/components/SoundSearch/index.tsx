import React, { useCallback, SyntheticEvent, ReactElement } from "react";
import "./index.css";
import { Nav, Form, InputGroup, Input } from "reactstrap";

export interface SoundSearchProps {
  onChange: { (query: string | null): void };
  query: string | null;
}

export default function SoundSearch({
  onChange,
  query
}: SoundSearchProps): ReactElement<SoundSearchProps> {
  const handleChange = useCallback(
    (event: SyntheticEvent<HTMLInputElement>) => {
      event.preventDefault();

      onChange(event.currentTarget.value);
    },
    [onChange]
  );

  return (
    <Nav navbar>
      <Form inline noValidate>
        <InputGroup>
          <Input
            value={query || ""}
            bsSize="sm"
            className="input-rounded"
            onChange={handleChange}
            placeholder="szukaj dżwięku"
            type="search"
          />
        </InputGroup>
      </Form>
    </Nav>
  );
}
