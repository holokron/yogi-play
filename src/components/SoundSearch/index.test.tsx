import * as React from "react";
import * as renderer from "react-test-renderer";
import SoundSearch, { SoundSearchProps } from ".";

describe("@components/SoundSearch", () => {
  it.each([
    [
      {
        onChange: (): void => {},
        query: null
      }
    ],
    [
      {
        onChange: (): void => {},
        query: "foo"
      }
    ]
  ] as SoundSearchProps[][])(
    "renders correctly with props: %p",
    (props: SoundSearchProps) => {
      const tree = renderer.create(<SoundSearch {...props} />).toJSON();

      expect(tree).toMatchSnapshot();
    }
  );
});
