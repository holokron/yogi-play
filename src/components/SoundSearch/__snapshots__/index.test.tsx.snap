// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/SoundSearch renders correctly 1`] = `
<ul
  className="navbar-nav"
>
  <form
    className="form-inline"
    noValidate={true}
  >
    <div
      className="input-group"
    >
      <input
        className="input-rounded form-control-sm form-control"
        onChange={[Function]}
        placeholder="szukaj dżwięku"
        type="search"
      />
    </div>
  </form>
</ul>
`;
